pkg/cri/sbserver/container_stats_list_linux.go

```diff
@@ -29,13 +29,55 @@ import (
 
 	containerstore "github.com/containerd/containerd/pkg/cri/store/container"
 	"github.com/containerd/containerd/pkg/cri/store/stats"
+	containerstorestats "github.com/containerd/containerd/pkg/cri/store/stats"
 )
 
 func (c *criService) containerMetrics(
 	meta containerstore.Metadata,
 	stats *types.Metric,
 ) (*runtime.ContainerStats, error) {
 	var cs runtime.ContainerStats
+	rawMetrics, err := c.rawContainerMetrics(meta, stats)
+	// If snapshotstore doesn't have cached snapshot information
+	// set WritableLayer usage to zero
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract container metrics: %w", err)
+	}
+	cs.Attributes = &runtime.ContainerAttributes{
+		Id:          rawMetrics.Attributes.Id,
+		Metadata:    rawMetrics.Attributes.Metadata,
+		Labels:      rawMetrics.Attributes.Labels,
+		Annotations: rawMetrics.Attributes.Labels,
+	}
+	cs.Cpu = &runtime.CpuUsage{
+		Timestamp:            rawMetrics.CPUStats.Timestamp,
+		UsageCoreNanoSeconds: &runtime.UInt64Value{Value: rawMetrics.CPUStats.UsageCoreNanoSeconds},
+		UsageNanoCores:       &runtime.UInt64Value{Value: rawMetrics.CPUStats.UsageNanoCores},
+	}
+	cs.Memory = &runtime.MemoryUsage{
+		Timestamp:       rawMetrics.MemoryStats.Timestamp,
```

Comment:
This can panic if there is an error getting memory stats and rawMetrics.MemoryStats is nil. One option is to check if MemoryStats is nil, but it may better to just embedded the inner structs directly (left another comment about this). This way we don't need to check for nil.